tests/datavolume_test.go

```
@@ -1254,6 +1276,30 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 					Message: "Import Complete",
 					Reason:  "Completed",
 				}}),
+			Entry("[test_id:5083]succeed importing VDDK data volume with extra arguments ConfigMap set", dataVolumeTestArguments{
```

Do you want to assert something in the final step of this test? like checking the config was applied?

---

For this test I wanted to make sure that the contents of the ConfigMap are present in the file, so the assertion happens in the vddk-test-plugin (the fgets/strcmp). Is there a better way to check the result from the importer? Like can this test read the pod logs or termination message?

---

oh okay I see https://github.com/kubevirt/containerized-data-importer/blob/625a9e944e3b75e514ca67e394c3e9931bf9634a/tools/vddk-test/vddk-test-plugin.c#L50 Yeah I think we can come up with a cleaner way,
You can either read pod logs or make this a part of the termination message struct https://github.com/kubevirt/containerized-data-importer/blob/625a9e944e3b75e514ca67e394c3e9931bf9634a/pkg/importer/vddk-datasource_amd64.go#L1011-L1019

---

WDYT?

---

I have an in-progress implementation to check the pod logs, but before I commit to that I am also going to try getting the result into the termination message to see if it's any cleaner. Either way it's definitely better to check the result from the test code than from the VDDK test plugin itself, that should get rid of a source of silent failures in CI.

---

The termination message approach should be cleaner, you would be asserting on a common.TerminationMessage struct that you got from the importer's API termination message

---

I tried this and putting the result in the termination message is only cleaner from this angle, the final assert. The rest of the implementation needed to have the importer scanning for test-only output from the nbdkit log just to save an arbitrary string value into the termination struct, which is fairly limited space that could probably be put to better use.

I think it ended up being nicer to just scan the log from the test, so the actual importer doesn't need to do extra stuff to accommodate the test.

---

Fair, thanks for trying :pray: